src/main/java/com/google/flourbot/api/Bot.java

```diff
@@ -50,7 +50,6 @@
   public static void main(String[] args) {
```

Comment:
L48: MacroExecutionModule macroExecutionModule (don't use implementations outside of the initializer and create descriptive variable names)
L63: nit: I think that "final String replyText;" will force the compiler to make sure the value is set in all paths of the switch.
L83: nit: Do not log PII (wtf/pii) I assume that name is user created name, always use machine generated ids.
L86: This is not a message that the user should see, throw IllegalArgumentException instead.
L96: The file name should be a const like chat scope for now. In the future it should come from some user defined config, so other developers that are using this code can easily customize this part (don't worry about it for now).

Reply:
Resolved: L48, L86

Reply:
Resolved: L96

Reply:
@lipczak-g when I changed replyText to "final String replyText;" I get an error saying "variable replyText might not have been initialized". I think this is because it is not assigned any value in the "REMOVED_FROM_SPACE" case. I don't think it makes sense to assign anything in this case because the bot has been removed from the room so it can't reply with anything anyways.

Reply:
oh, sorry for missing that, in such case the trick will not work. In such case I think we should add is empty check before it is used and throw IllegalArgumentException if it has not been initialized. It is always better to throw an exception instead of returning obviously incorrect result to the caller.

Reply:
Update: resolved all.